packages/fast-components-react-base/src/listbox-item/listbox-item.tsx

```diff
@@ -24,25 +25,60 @@ class ListboxItem extends Foundation<
         disabled: false,
     };
 
+    /**
+     * Gets the id to use for the item
+     * Authors who's items all have unique values can choose to
+     * not provide an itemId and the values of the items will be used as id's,
+     * otherwise use itemId.  All listbox items of a particular listbox should use the same
+     * approach with respect to using values or itemId or results could be unexpected.
+     * The "id" prop is being depracated and should not be used.
+     */
+    public static getItemId = (props: ListboxItemProps): string => {
+        return !isNil(props.itemId)
+            ? props.itemId
+            : !isNil(props.id)
+                ? props.id
+                : props.value;
+    };
+
+    /**
+     * Gets the id to use for an listbox item by examining the associated Element
+     */
+    public static getItemIdFromElement = (element: Element): string => {
+        let idToUse: string = element.getAttribute("itemId");
+        if (!isNil(idToUse)) {
+            return idToUse;
+        }
+
+        idToUse = element.getAttribute("id");
+        if (!isNil(idToUse)) {
+            return idToUse;
+        }
+
+        return element.getAttribute("value");
+    };
+
     protected handledProps: HandledProps<ListboxItemHandledProps> = {
         disabled: void 0,
         displayString: void 0,
         managedClasses: void 0,
         id: void 0,
+        itemId: void 0,
         onInvoke: void 0,
         value: void 0,
     };
 
     /**
      * Renders the component
      */
+
     public render(): React.ReactElement<HTMLDivElement> {
         return (
             <div
                 {...this.unhandledProps()}
                 className={this.generateClassNames()}
                 role="option"
-                id={this.props.id}
+                id={this.props.id || null}
```

Comment:
You shouldn't need to do this, if `this.props.id` is undefined I believe it just wont be added.